### ISSUE TITLE
Pill 8: Section 8.1 "inherit binutils-unwrapped" doesn't yield an environment variable in the builder

### ISSUE THREAD

**Issue by typetetris** [NixOS/nix-pills#43]
If you place `declare -xp` after setting the `PATH` in `hello_builder.sh` you can see

```
these derivations will be built:
  /nix/store/qy858zdhszinqwwcangw12pa9qlrkkm5-hello.drv
building path(s) ‘/nix/store/y47ll00ggfhprn5f7fm4nzaxp3pfvnaa-hello’
declare -x HOME="/homeless-shelter"
declare -x NIX_BUILD_CORES="4"
declare -x NIX_BUILD_TOP="/tmp/nix-build-hello.drv-0"
declare -x NIX_STORE="/nix/store"
declare -x OLDPWD
declare -x PATH="/nix/store/yph5q0shcy0x524c2088dpqbprdfbzjz-gnutar-1.30/bin:/nix/store/i9kmd5fpay17n2dfxzy3nx4pz98qb00a-gcc-wrapper-6.4.0/bin:/nix/store/jkc61h5hhs34h224609p9yl71r0yl7ki-gnumake-4.2.1/bin:/nix/store/9mq51i06492rn8ihavf0ff9j38pldg68-coreutils-8.29/bin:/nix/store/p8z0qyjhc61r0r60l9ahf6nrk2bpj7v6-gawk-4.2.0/bin:/nix/store/06h0afi1n99l8266nzp7617sq9kf7mv0-gzip-1.8/bin:/nix/store/a96ckwqkcjpibqvfgwqi4v0pjlizba8j-gnugrep-3.1/bin:/nix/store/fr8lipk5cnba99ikprxr2q0fbnxsp2r8-gnused-4.4/bin:/bin"
declare -x PWD="/tmp/nix-build-hello.drv-0"
declare -x SHLVL="1"
declare -x TEMP="/tmp/nix-build-hello.drv-0"
declare -x TEMPDIR="/tmp/nix-build-hello.drv-0"
declare -x TMP="/tmp/nix-build-hello.drv-0"
declare -x TMPDIR="/tmp/nix-build-hello.drv-0"
declare -x builder="/nix/store/nkq0n2m4shlbdvdq0qijib5zyzgmn0vq-bash-4.4-p12/bin/bash"
declare -x coreutils="/nix/store/9mq51i06492rn8ihavf0ff9j38pldg68-coreutils-8.29"
declare -x gawk="/nix/store/p8z0qyjhc61r0r60l9ahf6nrk2bpj7v6-gawk-4.2.0"
declare -x gcc="/nix/store/i9kmd5fpay17n2dfxzy3nx4pz98qb00a-gcc-wrapper-6.4.0"
declare -x gnugrep="/nix/store/a96ckwqkcjpibqvfgwqi4v0pjlizba8j-gnugrep-3.1"
declare -x gnumake="/nix/store/jkc61h5hhs34h224609p9yl71r0yl7ki-gnumake-4.2.1"
declare -x gnused="/nix/store/fr8lipk5cnba99ikprxr2q0fbnxsp2r8-gnused-4.4"
declare -x gnutar="/nix/store/yph5q0shcy0x524c2088dpqbprdfbzjz-gnutar-1.30"
declare -x gzip="/nix/store/06h0afi1n99l8266nzp7617sq9kf7mv0-gzip-1.8"
declare -x name="hello"
declare -x out="/nix/store/y47ll00ggfhprn5f7fm4nzaxp3pfvnaa-hello"
declare -x src="/nix/store/svc70mmzrlgq42m9acs0prsmci7ksh6h-hello-2.10.tar.gz"
declare -x system="x86_64-linux"
```

There is no environment variable starting with `binutils` and `PATH` ends in `/bin` which is the result of expand `${binutils_unwrapped}/bin` by bash.

Trying to set a shell parameter containing a `-` will be interpreted as a command by bash on my machine:

```
nix/store/nkq0n2m4shlbdvdq0qijib5zyzgmn0vq-bash-4.4-p12/bin/bash --version
GNU bash, version 4.4.12(1)-release (x86_64-unknown-linux-gnu)
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
```

and

```
test-uiaeue=uiae
bash: test-uiaeue=uiae: command not found
```
but `binutils_unwrapped` wasn't set also, as can be seen by the first output.

**Comment by Ericson2314** [comment]
I should take a look at this.

**Comment by DataDrover** [comment]
Just removing `_unwrapped`  from hello_builder.sh  and `-unwrapped` from hello.nix  seems to fix 8.1 (running Hummingbird).

**Comment by vitiral** [comment]
@DataDrover that does not work on the unstable branch. See the linked issue above. I had to do:

1. add `bintools = binutils.bintools;` to `hello.nix`
2. add `:$bintools/bin` to `PATH` in `build_hello.sh`

**Comment by vitiral** [comment]
the `autotools.nix` script had to be changed as well, here is mine:

```
pkgs: attrs:
  with pkgs;
  let 
    bintools = binutils.bintools;
    defaultAttrs = {
      builder = "${bash}/bin/bash";
      args = [ ./builder.sh ];
      baseInputs = [ gnutar gzip gnumake gcc bintools coreutils gawk gnused gnugrep ];
      buildInputs = [];
      system = builtins.currentSystem;
    };
  in
  derivation (defaultAttrs // attrs)

```

**Comment by DanBurton** [comment]
+1 @vitiral 's fixes worked for me on nixos 18.09
